common/src/main/java/com/wynntils/features/players/CustomNametagRendererFeature.java

```diff
@@ -198,6 +206,41 @@ private void drawNametags(PlayerNametagRenderEvent event, List<CustomNametag> na
                     nametag.nametagScale(),
                     yOffset);
         }
+
+        drawBadges(event, yOffset);
+    }
+
+    private void drawBadges(PlayerNametagRenderEvent event, float height) {
```

Comment:
if you always call height with 0 might as well just remove that parameter? also remove the if height == 0.

Or, is there any reason for it?

Reply:
If "show gear on hover" is enabled, the nametag gets shifted up to fit in the extra text. The badges need to be shifted up with it, so `height` stores the height of the gear nametags. I don't know why the if statement is necessary, but it works and there's something similar in legacy. My guess is the scale for the other nametags messes up the coordinates somehow.

Reply:
But height is always 0?

Reply:
What makes you say that? In `drawNametags`, yOffset is increased for every nametag drawn and is then passed in for height.

Reply:
Hm. I didn't see the line 211 call, only the 99 one. My bad.